haskell-best-practices.md

```diff
@@ -13,6 +13,67 @@
      x `divEx` d = x / d
  ```
 
+### Avoid `MonadFail`
+
+Don't write our own functions using `MonadFail`/`fail`.
+
+**Bad**:
+
+```hs
+data Email = Email
+
+parseEmail :: MonadFail m => Text -> m Email
+parseEmail = undefined
```

Comment:
can we use `_` instead of `undefined`. Or even write a simple body?

```
parseEmail :: MonadFail m => Text -> m Email
parseEmail
  | isEmail txt = Email txt
  | otherwise = fail "that isn't an email!"
```